metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/xml/DomQueryUsingXPath.java

```diff
@@ -24,19 +24,27 @@
 import javax.xml.parsers.DocumentBuilder;
 import javax.xml.parsers.DocumentBuilderFactory;
 import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
 import javax.xml.xpath.XPathFactory;
 import java.io.StringReader;
 
 public class DomQueryUsingXPath {
 
-    private final XPath xpath = XPathFactory.newInstance().newXPath();
+    private final XPath xPath = XPathFactory.newInstance().newXPath();
 
-    public Document buildDom(String xml) throws Exception {
+    private Document document;
+
+    public void buildDom(String xml) throws Exception {
```

Comment:
Given that this call is now absolutely required to guarantee the correct behavior of the `DomQueryUsingXPath`, make it part of the constructor instead of a separate method. That will ensure the document is not null when `isExpressionExists` is called.

This means that the `DomQueryUsingXPath` instance will no longer be provided externally to the `SraEntryXmlParser` constructor, but given that it will never be used outside that context, it doesn't really look like a bad thing :thinking: